### ISSUE TITLE
Create a CLI for `scancode-server`

### ISSUE THREAD

**Issue by chetanya-shrimali** [nexB/scancode-server#114]
create a command-line interface for scancode-server. So that we could run commands directly from the terminal.

**Comment by pombredanne** [comment]
Interesting idea! Can you elaborate a bit what would be the workflow? Especially assuming that the cli is not co-located with the server.

**Comment by yashLadha** [comment]
@pombredanne we can create a package for the project, using setuptools and Click/Docopt for creating the interface. Views @chetanya-shrimali

**Comment by pombredanne** [comment]
@yashLadha  exactly. I think I prefer Click (which is what is used in the ScanCode toolkit). One thing that would need to be fleshed out first would be what this CLI would offer?

**Comment by yashLadha** [comment]
@pombredanne  what i think is to include main functionalities like parsing and major functional endpoints in the code. That way the documentation of the CLI would be easy and meaningful.  Porting the whole code and giving meaningful endpoints for different function would be a good code. However i haven't gone through the codebase yet (suggesting on my past experience with CLI apps creation), i will be in better position to tell functional endpoints when i will atleast walk through the codebase.

**Comment by pombredanne** [comment]
@yashLadha ok, if you were to kinda sketch out what would the help of this CLI look like, what would it be roughly?

**Comment by yashLadha** [comment]
I have created a CLI for a web app, so if i think in that way then the help should contains the short param (`-`) and long param (`--`) and them all the necessary functional endpoints (file path or directory path to parse, custom config param or file, threads something like that). These params differ from project to project, so i won;t be able to solidify the proposal till i atleast walk through the code. @pombredanne . What is in your mind?

**Comment by pombredanne** [comment]
@yashLadha I meant by this: what would be the argument and options? what could you do with the CLI?

**Comment by pombredanne** [comment]
@yashLadha ping?

**Comment by yashLadha** [comment]
For that I have to look into the codebase. Currently my in-semester
examinations are going on.
